### PR TITLE
[VEN-1736]: RiskFund/PSR/Shortfall stage 1 VIP

### PR DESCRIPTION
**Pull request by kkirka**
This PR adds a simulation for the first stage of RiskFund, ProtocolShareReserve and Shortfall deployment. The VIP accepts ownership over the new contracts, configures access controls, and sets the Shortfall and ProtocolShareReserve addresses in VToken contracts.

Resolves VEN-1736

### CODE REVIEW

File: simulations/vip-160/vip-160/simulation.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  const markets = await Promise.all(
+    marketAddresses.map((address: string) => ethers.getContractAt(VTOKEN_ABI, address)),
+  );
+  return markets;
+};
+
+forking(30903270, () => {
+  let poolRegistry: Contract;
+
+  testVip("Risk fund, shortfall, PSR, stage 1", vip160(), {
```
Comment (chechu) [inline simulations/vip-160/vip-160/simulation.ts:46]: We could verify the emitted events

File: simulations/vip-160/vip-160/simulation.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  const marketAddresses = (
+    await Promise.all(comptrollers.map(async (comptroller: Contract) => comptroller.getAllMarkets()))
+  ).flat();
+  const markets = await Promise.all(
+    marketAddresses.map((address: string) => ethers.getContractAt(VTOKEN_ABI, address)),
+  );
+  return markets;
+};
+
+forking(30903270, () => {
```
Comment (chechu) [inline simulations/vip-160/vip-160/simulation.ts:43]: I would set a block number greater then 31340917, when the TWT market was added to the DeFi pool

File: simulations/vip-160/vip-160/simulation.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      }
+    });
+
+    it("sets Shortfall contract address for all markets", async () => {
+      const markets = await getAllMarkets(poolRegistry);
+      for (const market of markets) {
+        expect(await market.shortfall()).to.equal(SHORTFALL);
+      }
+    });
+  });
```
Comment (chechu) [inline simulations/vip-160/vip-160/simulation.ts:165]: Would it make sense to test the happy path, invoking the `reduceReserves` function on the VToken contracts, and them the `releaseFunds`, to finally check the amounts in the Treasury and the RiskFund contract? I know this is part of the test suite, so, maybe it's not needed. Wdyt?

File: vips/vip-160/vip-160.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+const NORMAL_TIMELOCK = "0x939bD8d64c0A9583A7Dcea9933f7b21697ab6396";
+const FAST_TRACK_TIMELOCK = "0x555ba73dB1b006F3f2C7dB7126d6e4343aDBce02";
+const CRITICAL_TIMELOCK = "0x213c446ec11e45b15a6E29C1C1b402B8897f606d";
+const PAUSE_GUARDIAN = "0x1C2CAc6ec528c20800B2fe734820D87b581eAA6B";
+const MANAGER = "0x7B1AE5Ea599bC56734624b95589e7E8E64C351c9";
+
+const SHORTFALL = "0x51dcdD4Ee45c0e8c04DE4C54908C2Ada38fd99B5";
+const RISK_FUND = "0xDF829fB0861DA07a0a935486738dFa65ED007927";
+const PROTOCOL_SHARE_RESERVE = "0x99731dDeA6e0d721dfeb434Cb20b200E72709056";
```
Comment (chechu) [inline vips/vip-160/vip-160.ts:14]: We need a PR with these addresses (and the testnet ones) in the docs repo

[the remark this answers is not in the document]
Comment (kkirka) [reply vips/vip-160/vip-160.ts:14]: Done

Comment (kkirka) [reply vips/vip-160/vip-160.ts:14]: https://github.com/VenusProtocol/venus-protocol-documentation/pull/76

File: vips/vip-160/vip-160.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  vHAY_Stablecoins: "0xCa2D81AA7C09A1a025De797600A7081146dceEd9",
+  vUSDT_Stablecoins: "0x5e3072305F9caE1c7A82F6Fe9E38811c74922c3B",
+  vUSDD_Stablecoins: "0xc3a45ad8812189cAb659aD99E64B1376f6aCD035",
+
+  vALPACA_DeFi: "0x02c5Fb0F26761093D297165e902e96D08576D344",
+  vANKR_DeFi: "0x19CE11C8817a1828D1d357DFBF62dCf5b0B2A362",
+  vUSDT_DeFi: "0x1D8bBDE12B6b34140604E18e9f9c6e14deC16854",
+  vBSW_DeFi: "0x8f657dFD3a1354DEB4545765fE6840cc54AFd379",
+  vUSDD_DeFi: "0xA615467caE6B9E0bb98BC04B4411d9296fd1dFa0",
+  vankrBNB_DeFi: "0x53728FD51060a85ac41974C6C3Eb1DaE42776723",
```
[the remark this answers is not in the document]
Comment (kkirka) [reply vips/vip-160/vip-160.ts:26]: [8f71988](https://github.com/VenusProtocol/vips/pull/59/commits/8f71988e009fa3315165e35d0d55e8c35e33cd08)

File: simulations/vip-160/vip-160/simulation.ts (hunk 2 above)
Comment (kkirka) [reply simulations/vip-160/vip-160/simulation.ts]: [8955ba0](https://github.com/VenusProtocol/vips/pull/59/commits/8955ba09b5cc5897eb7591a07e46a0590cd989a4)

File: simulations/vip-160/vip-160/simulation.ts (hunk 3 above)
Comment (kkirka) [reply simulations/vip-160/vip-160/simulation.ts:237]: [c038a61](https://github.com/VenusProtocol/vips/pull/59/commits/c038a61b682c6cc16b1a45aa90f76eb19f2bf6a4)

File: simulations/vip-160/vip-160/simulation.ts (hunk 1 above)
Comment (kkirka) [reply simulations/vip-160/vip-160/simulation.ts:78]: [c038a61](https://github.com/VenusProtocol/vips/pull/59/commits/c038a61b682c6cc16b1a45aa90f76eb19f2bf6a4)

**Review by chechu — APPROVED** [review]

**Review by chechu — APPROVED** [review]

**Comment by chechu** [comment]
https://app.venus.io/#/governance/proposal/170

**Pull request merged**